### PR TITLE
Add siren filter

### LINKED ISSUE #72: Handle siren requests with filter in Elasticsearch

### PR DESCRIPTION
**Pull request by HAEKADI**
closes #72

### CODE REVIEW

File: aio/aio-proxy/aio_proxy/search/search_functions.py [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 from elasticsearch_dsl import query
 
 
 def search_text(index, offset: int, page_size: int, **params):
     query_terms = params["terms"]
     s = index.search()
+
+    # Filter by siren first (if query is a `siren` number, and return search results
+    # directly.
+    if is_siren(query_terms):
```
Comment (geoffreyaldebert) [inline aio/aio-proxy/aio_proxy/search/search_functions.py:13]: No analysis for other params before ? 
If there is no params, go for it but if there are maybe it is another kind of search (eg for dirigeants for instance). 
My opinion is to be sure that there is only siren in query. If there is other kind of informations (in params). Don't do that.

File: aio/aio-proxy/aio_proxy/search/search_functions.py [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+        query_terms_clean = query_terms.replace(" ", "").upper()
```
Comment (geoffreyaldebert) [inline aio/aio-proxy/aio_proxy/search/search_functions.py:14]: upper is not useful right ? We normally have only numbers

File: aio/aio-proxy/aio_proxy/tests/unit_tests/test_is_siren.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from aio_proxy.search.helpers import is_siren
+
+
+@pytest.mark.parametrize(
+    "siren, expected",
+    [
+        ("123456789", True),
+        ("123 456 689", True),
+        (None, None),
+        ("mother of dragons", False),
```
Comment (geoffreyaldebert) [inline aio/aio-proxy/aio_proxy/tests/unit_tests/test_is_siren.py:11]: mother of dragons can not get `False` as a response but.. ok.

**Review by geoffreyaldebert — CHANGES_REQUESTED** [review]

File: aio/aio-proxy/aio_proxy/search/search_functions.py (hunk 1 above)
Comment (HAEKADI) [reply aio/aio-proxy/aio_proxy/search/search_functions.py:13]: I did discuss this case with @XavierJp yesterday, and we concluded that if a siren number is entered, more often than not no other filters are used.

File: aio/aio-proxy/aio_proxy/tests/unit_tests/test_is_siren.py (hunk 3 above)
Comment (HAEKADI) [reply aio/aio-proxy/aio_proxy/tests/unit_tests/test_is_siren.py:11]: Indeed 😂

File: aio/aio-proxy/aio_proxy/search/helpers.py [hunk 4]
```diff
@@ -32,6 +33,19 @@ def get_current_color(color_url):
 CURRENT_COLOR = get_current_color(os.getenv("COLOR_URL"))
 
 
+def is_siren(query_string: str):
+    """
+    Check if string is siren (composed of 9 digits).
+    """
+    if query_string is None:
+        return None
```
Comment (XavierJp) [inline aio/aio-proxy/aio_proxy/search/helpers.py:41]: Signature is Boolean here, shouldnt `None` return `False` ?

File: aio/aio-proxy/aio_proxy/tests/unit_tests/test_is_siren.py (hunk 3 above)
Comment (XavierJp) [reply aio/aio-proxy/aio_proxy/tests/unit_tests/test_is_siren.py:11]: You could add a siret here as the regex can be tricky between 9 and 14 digits

**Review by XavierJp — CHANGES_REQUESTED** [review]

File: aio/aio-proxy/aio_proxy/search/search_functions.py (hunk 1 above)
Comment (geoffreyaldebert) [reply aio/aio-proxy/aio_proxy/search/search_functions.py:13]: ok let's do it that way then

**Review by geoffreyaldebert — APPROVED** [review]

**Review by XavierJp — APPROVED** [review]

**Pull request merged**